baybe/searchspace/continuous.py

```diff
@@ -42,6 +44,36 @@ class SubspaceContinuous(SerialMixin):
     )
     """List of linear inequality constraints."""
 
+    def __repr__(self) -> str:
```

Comment:
Hey @AdrianSosic, regarding your last comment about overriding the `__str__` instead of `__repr__`: 
I tried it, but the problem was that `print(campaign)` did not call my new defined `__str__`, instead it called the standard one, and in some cases it did not print anything. Therefore, I just used `__repre__` instead.

Reply:
Hi @RimRihana. Not sure what exactly you tried, but you need to set things up such that they work nicely with `attrs`. Haven't looked too much into the `attrs` printing stuff but from what I've seen, I conclude that `attrs` uses the `__repr__` methods of nested classes to collect information about the class attributes and then assembles these into the default `__str__` method. And without further consideration, this also makes sense to me, because the `__repr__` is the stuff that contains full information about an object, while the print is arbitrary (hence, information collection needs to happen via `__repr__` of the nested classes, not `__str__`).

Long story short:
* You cannot simply override the `__repr__` instead of the `__str__` because that's not what we intend to do. The former needs to remain unchanged, but we want custom logic for the latter one.
* If you only modify the `__str__` methods of the nested classes, you will only see an effect for printing those classes, but not for classes that contain them as attributes
* Hence, you need to change the `__str__` of the nested classes and then explicitly call these new string methods on the highler levels (perhaps there is a more elegant way but I haven't had time to look into it)

Reply:
So if I understand this correctly: Probably, the only change (besides changin `__repr__` to `__str__`) would then be that we would not yet see any effect on levels that have search spaces as attributes? That is, we would see pretty printing for search spaces if we print them directly, but not if we print them as a part of a campaign (unless we override the `__str__` there)?
That feels a bit weird and unintuitive for me, I would have assumed that `attrs` would just glue the `__str__` methods together. But I guess this is not too big of an issue, since campaign will probably just call of the attribute's printing functions and glue them together, right? There won't be much to format.

Reply:
Thank you @AdrianSosic for explaining this! Now I know why it did not work before. Probably because I was trying to print the campaign object and implicitly called `__repr__` instead of `__str__`. 
I will override `__str__` instead of `__repr__` 😃